pkgs/applications/version-management/dvc/dvc-daemon.patch

```diff
@@ -0,0 +1,21 @@
+diff --git a/dvc/daemon.py b/dvc/daemon.py
+index 1d67a37..7ce6fde 100644
+--- a/dvc/daemon.py
++++ b/dvc/daemon.py
+@@ -67,14 +67,8 @@ def daemon(args):
+     Args:
+         args (list): list of arguments to append to `dvc daemon` command.
+     """
+-    cmd = [sys.executable]
+-    if not is_binary():
+-        cmd += ['-m', 'dvc']
+-    cmd += ['daemon', '-q'] + args
+-
+-    env = fix_env()
+-    file_path = os.path.abspath(inspect.stack()[0][1])
+-    env['PYTHONPATH'] = os.path.dirname(os.path.dirname(file_path))
++    cmd = ["dvc", "daemon", "-q"] + args
```

Comment:
Please do make use of `$out`. That way, it will also work as expected if multiple versions of dvc are installed.

Reply:
How do you do that in a patch?

Reply:
Does that work with patch files?

Reply:
Yes, you just have to put a placeholder, e.g. `@dvc@` in the patch.

Reply:
In `postPatch`.

Reply:
Just to clarify. Does this mean we patch, then substitute, or substitute the patch and then apply the patch?

Reply:
In patch
```
@dvc@ # a placeholder for substituteAll
```

in expression
```
patches = [
  (substituteAll {
    src = ./path_to_patch;
    dvc = "${placeholder "out"}/bin/dvc";
  })
];

postPatch = ''
  substituteInPlace path_to_file_that_was_patched_dvc_placeholder \
    --subst-var-by $out
'';
```

> example https://github.com/NixOS/nixpkgs/blob/master/pkgs/desktops/gnome-3/core/evolution-data-server/default.nix#L27

Reply:
Can we do something like this: 
In default.nix :
add
`postPatch = ''
    substituteInPlace dvc/daemon.py --subst-var-by dvc "$out/bin/dcv"
  '';`

in patch file
modify
`cmd = [ *"dvc"* , "daemon", "-q"] + args to cmd = [ *"@dvc@"* , "daemon", "-q"] + args`

Reply:
Seems like it should be easier to just substitute into the patch directly and then apply the patch as suggested by @Rakesh4G? What are your thoughts @worldofpeace @dotlambda

Reply:
Either one is fine, I just happen to use the first option we suggested all over nixpkgs.